### PR TITLE
moved HideAllItems Tag from AdvancedSettings to GUI

### PR DESCRIPTION
**Pull request by mad-max**
Hey Guys,

based on my old PR #819 and the discussion on the forums ( http://forum.kodi.tv/showthread.php?tid=207425 ) I reworked the old attempt to move "HideAllItems" from advancedsettings.xml to the GUI.

Opinions?

cheers,
mad-max

### CODE REVIEW

File: language/English/strings.po [hunk 1]
```diff
@@ -16254,3 +16254,12 @@ msgstr ""
 msgctxt "#38010"
 msgid "GPU accelerated"
 msgstr ""
+
+#: system/settings/settings.xml
+msgctxt "#38011"
+msgid "Hide All Items entry"
```
Comment (Montellese) [inline language/English/strings.po]: Maybe add (escaped) double-quotes around `All Items` to make it more obvious. (same in the description).

File: language/English/strings.po [hunk 2]
```diff
@@ -16254,3 +16254,12 @@ msgstr ""
 msgctxt "#38010"
 msgid "GPU accelerated"
 msgstr ""
+
+#: system/settings/settings.xml
```
Comment (Montellese) [inline language/English/strings.po]: This is missing the context description.

File: language/English/strings.po [hunk 3]
```diff
@@ -16254,3 +16254,12 @@ msgstr ""
...
+msgctxt "#38011"
+msgid "Hide All Items entry"
+msgstr ""
+
+msgctxt "#38012"
```
Comment (Montellese) [inline language/English/strings.po]: This is missing the file reference and the context description.

File: xbmc/settings/Settings.cpp [hunk 4]
```diff
@@ -666,6 +666,7 @@ void CSettings::InitializeISettingCallbacks()
   settingSet.clear();
   settingSet.insert("karaoke.export");
   settingSet.insert("karaoke.importcsv");
+  settingSet.insert("musiclibrary.hideallitems");
```
Comment (Montellese) [inline xbmc/settings/Settings.cpp]: You don't need these as you don't do anything explicitly (i.e. you don't handle them in one of the `ISettingCallback` callbacks) if the settings are changed.

**Comment by Montellese** [comment]
Looks good to me (apart from the minors) and IMO makes sense. It's a real shame that we can't migrate the values of advanced settings :-/

File: language/English/strings.po (hunk 1 above)
Comment (mad-max) [inline language/English/strings.po]: Thought about this, too. I wasn't sure about it, but I can update

**Comment by mad-max** [comment]
Let me get the minors sorted...maybe there are a few other things that can move from as.xml to the GUI

**Comment by Montellese** [comment]
@mad-max: Sort out the minors and then we get this in. I'd do this on a one-by-one basis as it makes it easier to review and it gets included faster as well.

**Comment by mad-max** [comment]
Yeah, that's what I meant ^^
Just a little work for after this PR ;)

**Comment by mad-max** [comment]
@Montellese minors should be fixed...
If changes are ok, I'll squash them

**Comment by Montellese** [comment]
Looks good for me. @MartijnKaijser @mkortstiege any objections?

**Comment by mkortstiege** [comment]
Too bad we can't promote as to gui settings :/ Other than that it looks good.

**Comment by MartijnKaijser** [comment]
all good 
jenkins build this please

**Comment by Montellese** [comment]
Hint: This needs to be squashed before it can be merged (which is why I didn't start a jenkins build).

**Comment by mad-max** [comment]
I waited for Matthias and Martijn before squashing.
When Jenkins is done, I'll squash as promised above...

**Comment by MartijnKaijser** [comment]
for some silly reason it failed on several platforms.
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

File: xbmc/filesystem/MusicDatabaseDirectory/DirectoryNode.cpp [hunk 5]
```diff
@@ -46,6 +46,7 @@
 #include "utils/StringUtils.h"
 #include "guilib/LocalizeStrings.h"
 #include "music/MusicDbUrl.h"
+#include "settings/settings.h"
```
Comment (Paxxi) [inline xbmc/filesystem/MusicDatabaseDirectory/DirectoryNode.cpp]: missed an uppercase S here, should be settings/Settings.h and that's why it's failing on linux/android

File: xbmc/filesystem/VideoDatabaseDirectory/DirectoryNode.cpp [hunk 6]
```diff
@@ -44,6 +44,7 @@
 #include "guilib/LocalizeStrings.h"
 #include "utils/Variant.h"
 #include "video/VideoDatabase.h"
+#include "settings/settings.h"
```
Comment (Paxxi) [inline xbmc/filesystem/VideoDatabaseDirectory/DirectoryNode.cpp]: same here missed an uppercase S here, should be settings/Settings.h and that's why it's failing on linux/android

**Comment by mad-max** [comment]
@MartijnKaijser 
I think you can try again ;)

**Comment by mkortstiege** [comment]
jenkins build this please

File: language/English/strings.po [hunk 7]
```diff
@@ -16254,3 +16254,15 @@ msgstr ""
 msgctxt "#38010"
 msgid "GPU accelerated"
 msgstr ""
+
+#. Setting #38011 "Videos -> Library -> Hide All Items entry"
+#: system/settings/settings.xml
+msgctxt "#38011"
+msgid "Hide "All Items" entry"
```
Comment (da-anda) [inline language/English/strings.po]: shouldn't the quotes within the qoutes be escaped somehow? @alanwww1 ?

**Comment by da-anda** [comment]
too bad I missed this. I would have preferred if the logic would have been inverted, so "show 'All items' entry" with it being enabled by default. But so what.

**Comment by MartijnKaijser** [comment]
@da-anda how are other things mentioned in hide or show (actually didn't pay attention to that)? If not consistent it should be reversed.

File: language/English/strings.po (hunk 7 above)
Comment (mad-max) [inline language/English/strings.po]: Tested this and it worked fine

**Comment by da-anda** [comment]
@MartijnKaijser IIRC we currently have a mix of "hide X" and "show X" options or "disable X" and "enable X". Ideally we would clean this up at some point, but this will require some migration code to drag along for quite some time

**Comment by MartijnKaijser** [comment]
ok so we need to decide a direction and stick with that. It's either show or hide so that would make it relative easy to decide.

**Comment by mad-max** [comment]
@da-anda yes, currently there is a mix and to be honest, I did not think about reversing it, which is IMO the better way, as you said

**Comment by Montellese** [comment]
IMO right now you can still reverse it without providing any migration code because it has only been in a few nightlies.

**Comment by da-anda** [comment]
@MartijnKaijser yes, we need to decide. IMO best is if a setting is "enabling" somthing when checked, so "show FOO" and "enable FOO".

@Montellese sure, shouldn't be an issue to change it now

**Comment by mad-max** [comment]
Reversed the logic an created new PR #6601

**Comment by MilhouseVH** [comment]
Just for the sake of consistency with Video -> Library, shouldn't the Music -> Library `Show "All items" entry` setting be grouped above the horizontal rule immediately above `Upload library on startup`? It seems a bit odd having it grouped with `Clean library...`

**Pull request merged**